libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -1214,7 +1220,6 @@ static CK_RV prvSaveDerKeyToPal( mbedtls_pk_context * pxMbedContext,
 
             if( xResult == CKR_OK )
             {
-                xFreeMemory = CK_TRUE;
```

Comment:
We're still unconditionally calling vPortFree below, though, even in the case where the malloc failed.

Reply:
Alfred pointed out that free handles the case when it is passed  a NULL pointer, so there is no reason to check if malloc failed when freeing